samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,241 @@
+    }
+
+    /**
+     * Assert that the file to be copied exists at the expected location.
+     * This method waits a duration which is defined in {@link FileTransferSampleTest#TIMEOUT}.
+     */
+    void assertWaitForDestinationFileExistence() {
+        await().atMost(TIMEOUT).pollInterval(POLL_INTERVAL).untilAsserted(() -> assertThat(destinationFile).exists());
+    }
+
+    /**
+     * Assert that a POST request to initiate a contract negotiation is successful.
+     * This method corresponds to the command in the sample: curl -X POST -H "Content-Type: application/json" -H "X-Api-Key: password" -d @samples/04.0-file-transfer/contractoffer.json "http://localhost:9192/api/v1/data/contractnegotiations"
+     */
+    void assertInitiateContractNegotiation() {
+        JsonPath jsonPath = RestAssured
+            .given()
+                .headers(API_KEY_HEADER_KEY, API_KEY_HEADER_VALUE)
+                .contentType(ContentType.JSON)
+                .body(new File(TestUtils.findBuildRoot(), CONTRACT_OFFER_FILE_PATH))
+            .when()
+                .post(INITIATE_CONTRACT_NEGOTIATION_URI)
+            .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body("id", not(emptyString()))
+                .extract()
+                .jsonPath();
+
+        contractNegotiationId = jsonPath.get("id");
+    }
+
+    /**
+     * Assert that a GET request to look up a contract agreement is successful.
+     * This method corresponds to the command in the sample: curl -X GET -H 'X-Api-Key: password' "http://localhost:9192/api/v1/data/contractnegotiations/{UUID}"
+     */
+    void assertLookUpContractAgreementId() {
+        var localContractAgreementId = new AtomicReference<String>();
+
+        // Wait for transfer to be completed.
+        await().atMost(TIMEOUT).pollInterval(POLL_INTERVAL).untilAsserted(() -> {
+                    var result = RestAssured
+                        .given()
+                            .headers(API_KEY_HEADER_KEY, API_KEY_HEADER_VALUE)
+                        .when()
+                            .get(LOOK_UP_CONTRACT_AGREEMENT_URI, contractNegotiationId)
+                        .then()
+                            .statusCode(HttpStatus.SC_OK)
+                            .body("state", equalTo("CONFIRMED"))
+                            .body("contractAgreementId", not(emptyString()))
+                            .extract().body().jsonPath().getString("contractAgreementId");
+
+                    localContractAgreementId.set(result);
+
+                }
+        );
+
+        contractAgreementId = localContractAgreementId.get();
+    }
+
+    /**
+     * Assert that a POST request to initiate transfer process is successful.
+     * This methods corresponds to the command in the sample: curl -X POST -H "Content-Type: application/json" -H "X-Api-Key: password" -d @samples/04.0-file-transfer/filetransfer.json "http://localhost:9192/api/v1/data/transferprocess"
```

Comment:
```suggestion
     * This method corresponds to the command in the sample: {@code curl -X POST -H "Content-Type: application/json" -H "X-Api-Key: password" -d @samples/04.0-file-transfer/filetransfer.json "http://localhost:9192/api/v1/data/transferprocess"}
```

similar for other instances